src/extensions/paramHelper.ts

```diff
@@ -16,6 +16,41 @@ import * as askUserForParam from "./askUserForParam";
 import * as track from "../track";
 import * as env from "../functions/env";
 
+/**
+ * Interface for holding different param values for different environments/configs.
+ */
+export interface ParamBindingOptions {
+  default: string;
```

Comment:
Very much like this ParamBindingOptions approach overall. One nit is that I'd prefer to use a different word than 'default' to represent the value provided in `instanceId.env` - default already has a different meaning in the context of params (the Param.default value). Consider something like `baseValue` instead?